packages/components/autocomplete/src/autocomplete.vue

```diff
@@ -280,10 +278,11 @@ const highlight = (index: number) => {
 onClickOutside(listboxRef, close)
 
 onMounted(() => {
-  inputRef.value!.inputOrTextarea.setAttribute('role', 'textbox')
-  inputRef.value!.inputOrTextarea.setAttribute('aria-autocomplete', 'list')
-  inputRef.value!.inputOrTextarea.setAttribute('aria-controls', 'id')
-  inputRef.value!.inputOrTextarea.setAttribute(
+  // TODO: use Volar generate dts to fix it.
+  ;(inputRef.value as any).ref!.setAttribute('role', 'textbox')
+  ;(inputRef.value as any).ref!.setAttribute('aria-autocomplete', 'list')
+  ;(inputRef.value as any).ref!.setAttribute('aria-controls', 'id')
+  ;(inputRef.value as any).ref!.setAttribute(
```

Comment:
**Suggestion:**

```ts
const inputElement = (inputRef.value as any).ref!
[
  { key: 'role', value: 'textbox'},
  { key: 'aria-autocomplete', value: 'list' },
  { key: 'aria-controls', value: 'id' },
  { key: 'aria-activedescendant', value: `${id.value}-item-${highlightedIndex.value}` }
].forEach(({ key, value }) => inputElement.setAttribute(key, value))

```

Reply:
<img width="1518" alt="image" src="https://user-images.githubusercontent.com/6481596/158023803-77f0de32-bcab-499b-a03a-659056947e38.png">

This is not the main refactoring component of this PR, and we can do it in another PR.